Classes/Service/LanguageService.php

```diff
@@ -56,74 +43,43 @@ public function getSourceLanguage(Site $currentSite): array
     }
 
     /**
-     * @return array{uid: int, title: string, language_isocode: string, languageCode: string, formality: string}
+     * @return array{uid: int, title: string, language_isocode: string, languageCode: string, formality: string}|false
      * @throws LanguageRecordNotFoundException
      * @throws LanguageIsoCodeNotFoundException
+     * @throws InvalidArgumentException
      */
-    public function getTargetLanguage(Site $currentSite, int $languageId): array
+    public function getTargetLanguage(Site $currentSite, int $languageId): array|false
     {
-        // @todo TYPO3 v12 changed locale API and therefore site configuration. Configured languages do no longer
-        //       directly contains values like hreflang or iso-639-1 directly. Possible workarounds would be to
-        //       operate directly on the siteLanguage objects and no longer use the raw configuration values.
-        //       See: https://review.typo3.org/c/Packages/TYPO3.CMS/+/77807
-        //            https://review.typo3.org/c/Packages/TYPO3.CMS/+/77597
-        //            https://review.typo3.org/c/Packages/TYPO3.CMS/+/77726
-        //            https://review.typo3.org/c/Packages/TYPO3.CMS/+/77814
-        $languages = array_filter($currentSite->getConfiguration()['languages'], function ($value) use ($languageId) {
-            if (!is_array($value)) {
-                return false;
-            }
-
-            if ((int)$value['languageId'] === $languageId) {
-                return true;
+        try {
+            $language = $currentSite->getLanguageById($languageId);
+        } catch (\Exception $e) {
+            if ($e->getCode() === 1522960188) {
+                throw new LanguageRecordNotFoundException(
+                    sprintf('Language "%d" in site "%s" not found.', $languageId, $currentSite->getIdentifier()),
+                    1746959505
```

Comment:
Original exception should be added as parent exception to get the full exception chain and a trace down to the whole exception stack.